src/permutations.jl

```diff
@@ -211,10 +211,22 @@ function Base.length(c::MultiSetPermutations)
     return round(Int, p[t+1])
 end
 
+
+"""
+    multiset_permutations(a)
+
+Generate all permutations of an array `a` where `a` may have duplicated elements.
+"""
+function multiset_permutations(a)
+    m = unique(collect(a))
+    f = [count(==(x), a) for x in m]
+    multiset_permutations(m, f, length(a))
+end
```

Comment:
```suggestion
multiset_permutations(a) = multiset_permutations(a, length(a))
```

just like:
https://github.com/JuliaMath/Combinatorics.jl/blob/2ad84aac51415fbc101700fc495c931353c324b7/src/permutations.jl#L93